Assets/HoloToolkit/Input/Scripts/Cursor/InteractiveMeshCursor.cs

```diff
@@ -153,15 +155,38 @@ public override void SetVisiblity(bool visible)
 
             if (visible)
             {
+                if (Light != null)
```

Comment:
I'm finding this a little harder to follow because some state is implemented in this function related to Light, some is updated in OnCursorStateChange, and some is in ElementVisibility. Which one is used depends on whether we're becoming visible or invisible. If I were modifying this code, I think I'd have a hard time convincing myself that I'm doing it correctly for all the possible cases. Would it be possible to simplify this so that all state updates flow through a single path?